lib/stellar_core_commander/transactor.rb

```diff
@@ -157,10 +156,32 @@ def get_process(name)
       end
     end
 
+    Contract Symbol, ArrayOf[Symbol], Num => Process
+    def process(name, quorum=[name], thresh=quorum.length)
+      $stderr.puts "creating process #{name}"
+      p = @commander.make_process self, name, quorum, thresh
+      $stderr.puts "process #{name} is #{p.idname}"
+      add_named name, p
+    end
+
+    Contract Symbol, Proc => Any
+    def on(process_name)
+      @commander.start_all_processes
+      p = get_process process_name
+      $stderr.puts "executing steps on #{p.idname}"
+      tmp = @process
+      begin
+        @process = p
+        yield
+      ensure
+        @process = tmp
+      end
+    end
```

Comment:
Like the example I gave with the `rescue` clause, `ensure` can also be tacked onto a method definition without need for a separate begin block:

```ruby
Contract Symbol, Proc => Any
def on(process_name)
  @commander.start_all_processes
  p = get_process process_name
  $stderr.puts "executing steps on #{p.idname}"
  tmp = @process
  @process = p
  yield
ensure
  @process = tmp
end
```